spec/services/apis/youtubes/create_video_service_spec.rb

```diff
@@ -0,0 +1,326 @@
+        TestResponseHash.hash[:test_alt][:snippet][:published_at]
+      end
+    end
+
+    class TestThumbnailDetailsAlt
+      def high
+        TestThumbnailAlt.new
+      end
+    end
+
+    class TestThumbnailAlt
+      def url
+        "test_thumbnail_alt"
+      end
+    end
+
+    class TestContentDetailsAlt
+      def duration
+        "test_duration_alt"
+      end
+    end
+
+    #################各クラス共有部分#################
+
+    class TestResponseHash
+      def self.hash
+        {
+          test:
+          {
+            id: "test_video_code",
+            snippet:
+            {
+              channel_id: "test_channel_code",
+              channel_title: "test_channel_title",
+              title: "test_video_title",
+              description: "test_description",
+              thumbnails: TestThumbnailDetails.new,
+              published_at: "Sun, 01 Jan 1111 00:00:00 +0000"
+            },
+            content_details: TestContentDetails.new
+          },
+          test_alt:
+          {
+            id: "test_video_code_alt",
+            snippet:
+            {
+              channel_id: "test_channel_code",
+              channel_title: "test_channel_title",
+              title: "test_video_title_alt",
+              description: "test_description_alt",
+              thumbnails: TestThumbnailDetailsAlt.new,
+              published_at: "Mon, 02 Feb 2222 00:00:00 +0000"
+            },
+            content_details: TestContentDetailsAlt.new
+          }
+        }
+      end
+    end
+
+    #################ここまでレスポンスのクラス定義#################
+
+    let(:video) { Apis::Youtubes::CreateVideoService.new }
+    let(:list_mock) { double("list_videos") }
+    before do
+      allow(video).to receive(:list).and_return(list_mock)
+    end
+
+    context "新規Channelが登録される場合" do
+      let(:test_items) { TestItem.new }
+      let(:code) { "test_video_code" }
+      before do
+        allow(list_mock).to receive(:items).and_return(test_items)
+      end
+
+      it "Channel及びVideoが保存されること" do
+        video.execute!(code)
+        test_channel = Channel.last
+        test_video = Video.last
+
+        aggregate_failures do
+          expect(test_channel.code).to eq "test_channel_code"
+          expect(test_channel.title).to eq "test_channel_title"
+          expect(test_video.code).to eq "test_video_code"
+          expect(test_video.title).to eq "test_video_title"
+          expect(test_video.description).to eq "test_description"
+          expect(test_video.thumbnail).to eq "test_thumbnail"
+          expect(test_video.duration).to eq "test_duration"
+          expect(test_video.published_at).to eq "Sun, 01 Jan 1111 00:00:00 +0000"
+          expect(test_video.channel_id).to eq test_channel.id
+        end
+      end
+    end
+
+    context "既存Channelに新規Videoが登録される場合" do
+      let(:test_items_alt) { TestItemAlt.new }
+      let(:code) { "test_video_code_alt" }
+      before do
+        @test_channel = Channel.create(
```

Comment:
```suggestion
        let!(:test_channel)  { create(:channel, xxxx) } 
```

@よりこっちのほうが良さげ。

ここのcreateは factory bot を調べてください。